src/cohort-manager.js

```diff
@@ -0,0 +1,132 @@
+const Cohort = require('./cohort')
+const Student = require('./student')
+
+class CohortManager {
+  constructor() {
+    this.cohortArray = []
+    this.studentID = 1
+  }
+
+  addCohort(name, maxCapacity) {
+    if (name.length === 0) {
+      return false
+    }
+    for (let i = 0; i < this.cohortArray.length; i++) {
+      const cohort = this.cohortArray[i]
+      if (name === cohort.name) {
+        return false
+      }
+    }
+    const newCohort = new Cohort(name, maxCapacity)
+    this.cohortArray.push(newCohort)
+    return this.cohortArray
+  }
+
+  removeCohort(name) {
+    for (let i = 0; i < this.cohortArray.length; i++) {
+      const cohort = this.cohortArray[i]
+      if (name === cohort.name) {
+        this.cohortArray.splice(i, 1)
+        return this.cohortArray
+      }
+    }
+    return false
+  }
+
+  searchByCohortName(name) {
+    for (let i = 0; i < this.cohortArray.length; i++) {
+      const cohort = this.cohortArray[i]
+      if (name === cohort.name) {
+        return cohort
+      }
+    }
+    return 'This cohort does not exist'
+  }
+
+  searchByStudentEmail(email) {
+    for (let i = 0; i < this.cohortArray.length; i++) {
+      for (let j = 0; j < this.cohortArray[i].studentsArray.length; j++) {
```

Comment:
How did you find nested loops? Was it confusing? Easy to use?

Reply:
I used nested loops on criteria too, but I found it confusing. For me, it was difficult to imagine and keep track of where the loop is doing, but I found it helpful to use describing letters for the LET variable. for example, a = 0 iterates on the array, and c = 0 iterates on the cohort.

Reply:
Try to visualize a board of a familiar game that uses numbers/letters to identify rows and columns, like battleships. Here the oustide loop (first one) loops through every row, and the inside loop (second one) loops through every column. So, starting at the first row, every column on that row is checked. Then it moves to the next rrow and repeats, and keeps doing this until there is no more rows to check.